agent/worker.go

```diff
@@ -484,3 +489,36 @@ func (w *worker) Subscribe(ctx context.Context, subscription *api.SubscriptionMe
 		}
 	}
 }
+
+// Leave removes all assignments
+func (w *worker) Leave(ctx context.Context) error {
+	var err error
+	w.Close()
+
+	if err = w.Assign(ctx, nil); err != nil {
+		return err
+	}
+
+	// wait and confirm tasks are removed from the node
+	for retries := 0; retries < 3; retries++ {
+		select {
+		case <-time.After(50 * time.Millisecond):
+			w.mu.Lock()
```

Comment:
No need to hold this lock. `w.db` can be accessed safely and `w.db.View` provides consistent view with transaction.